api/receiver/azure.go

```diff
@@ -35,65 +46,80 @@ func newAzureArtifact(artifactName string, zip io.Reader) (azureArtifact, error)
 	}, err
 }
 
-// extractReports extracts report files from the given zip.
-func (a azureArtifact) getReportFiles() ([]*zip.File, error) {
+// getFilesMatchingPattern returns the number and a slice of *zip.File whose
+// filenames match the given pattern.
+func (a azureArtifact) getFilesMatchingPattern(p *regexp.Regexp) (int, []*zip.File, error) {
 	if a.z == nil {
 		r, err := zip.NewReader(bytes.NewReader(a.data), int64(len(a.data)))
 		if err != nil {
-			return nil, err
+			return 0, nil, err
 		}
 		a.z = r
 	}
 	files := make([]*zip.File, 0, len(a.z.File))
 	for _, f := range a.z.File {
-		if reportPathRegex.MatchString(f.Name) {
+		if p.MatchString(f.Name) {
 			files = append(files, f)
 		}
 	}
-	return files, nil
+	return len(files), files, nil
 }
 
-func getAzureArtifactName(url string) string {
-	if match := azureArtifactRegex.FindStringSubmatch(url); len(match) > 1 {
-		return match[1]
+// gzipReaderFromZip returns a *function* that returns the gzipped i-th file in
+// the given slice of *zip.File (the return value is intended to be used by
+// sendResultsToProcessor).
+func gzipReaderFromZip(files []*zip.File) func(int) (io.ReadCloser, error) {
+	return func(i int) (io.ReadCloser, error) {
+		zipR, err := files[i].Open()
+		if err != nil {
+			return nil, err
+		}
+		buf := new(bytes.Buffer)
+		gzipW := gzip.NewWriter(buf)
```

Comment:
Seems strange to force the content back into gzip just to be read out again?

Reply:
Yeah this is not ideal. It's the same as what the code currently does, which is because in a later part of the receiver, the file is unconditionally treated as gzipped (there's a TODO to detect whether the file is gzipped). Let me also add a TODO here.